cmake/modules/OmrDetectSystemInformation.cmake

```diff
@@ -193,10 +193,21 @@ macro(omr_detect_system_information)
 			if("${CMAKE_C_SIMULATE_ID}" STREQUAL "MSVC" OR "${CMAKE_CXX_SIMULATE_ID}" STREQUAL "MSVC")
 				# clang on Windows mimics MSVC
 				set(_OMR_TOOLCONFIG "msvc")
-			else()
-				# TODO we don't actually have a clang config
-				# just use GNU config
-				set(_OMR_TOOLCONFIG "gnu")
+			elseif(CMAKE_C_COMPILER_ID MATCHES "^Clang$")
```

Comment:
Z system will use OpenXL compiler too.  Have you make sure either of the two below?
1) Z system will not use Clang compiler ID for that expected switch;  or
2) this config setting is applicable to them as well.

Reply:
@r30shah please comment ...

Reply:
Yes, I believe the COMPILER_ID on Z will be `IBMClang` and TOOLCONFIG will be set to `openxl`. I would like @Deigue to confirm this.

Reply:
I didn't consider that the COMPILER_ID can be different for Z system.

Reply:
you had better get together with @Deigue
either,  both can use the same ID, if we can; 
or, AIX uses a more differentiating and meaningful ID, rather than this very generic "Clang"

Reply:
I can't find any existing toolconfig for "openxl" that's why I used "gnu" for now.

Reply:
I've made the compiler id related changes here: https://github.com/eclipse/omr/pull/7319 ,
also will be moving the `"IBMClang$"` somewhere alongside the xlc clause so that all tha compiler id variations can navigate to openxl toolchain easily from one spot based on Keiths feedback.

From what I remember, I was seeing "ibm-clang" as the compiler id while running locally, but when trying a Open XL build on Jenkins for z/OS, I was seeing the compiler id come out as "IBMClang" ...

In any case, I did create my own "openxl.cmake" toolconfig in the PR above.

Reply:
@Deigue Can this openxl.cmake be used for AIX as well?

Reply:
I would say yes, because there is a section in the file I left behind that says
`if(OMR_OS_AIX)` , that leaves the default state of setup/configs, I suspect from `xlc.cmake` .

So you would be able to use that section, but you may need to adapt the AIX configs to be whatever you need to make things work.